validation-test/stdlib/String.swift

```diff
@@ -2190,4 +2190,10 @@ StringTests.test("NormalizationBufferCrashRegressionTest") {
   expectTrue(set.contains(str))
 }
 
+StringTests.test("NormalizationSanityCheck") {
+  let str = "\u{0336}\u{0344}\u{0357}\u{0343}\u{0314}\u{0351}\u{0340}\u{0300}\u{0340}\u{0360}\u{0314}\u{0357}\u{0315}\u{0301}\u{0344}a"
+  let strNFC = "\u{0336}\u{0308}\u{0301}\u{0357}\u{0313}\u{0314}\u{0351}\u{0300}\u{0300}\u{0300}\u{0314}\u{0357}\u{0301}\u{0308}\u{0301}\u{0315}\u{0360}\u{0061}"
+  expectEqual(str, strNFC)
+}
+
```

Comment:
Were you able to test the case where the previous approach would've skipped over the long segment?

Reply:
Yes, this new test caught it.

Reply:
The concern was that a corner case would cause us to skip a portion of the String. Unfortunately, a test case expressed in terms of string comparison is only testing behavioral consistency but not the actual result. We have `_nfcCodeUnits` and `_withNFCCodeUnits` to check exactly this.